File: GRMCore/Class/cSetCrossSection.vb

Public MustInherit Class cSetCrossSection

    Public Enum CSTypeEnum
        CSSingle
        CSCompound
    End Enum

    Public MustOverride ReadOnly Property CSType() As CSTypeEnum
    Public MustOverride Sub GetValues(ByVal prjds As GRMProject)
    Public MustOverride ReadOnly Property IsSet() As Boolean
    Public MustOverride Sub SetValues(ByVal prjds As GRMProject)

End Class

File: GRMCore/Class/cSetCrossSection.cs

namespace GRMCore
{
    public abstract class cSetCrossSection
    {
        public enum CSTypeEnum
        {
            CSSingle,
            CSCompound
        }

        public abstract CSTypeEnum CSType { get; }
        public abstract void GetValues(Dataset.GRMProject prjds);
        public abstract bool IsSet { get; }
        public abstract void SetValues(Dataset.GRMProject prjds);
    }

}
